src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -93,6 +93,49 @@ public class MatchInfo
         /// <value>The text of the matching line.</value>
         public string Line { get; set; } = string.Empty;
 
+        /// <summary>
+        /// Gets or sets a value indicating whether the matched portion of the string is highlighted.
+        /// </summary>
+        /// <value>Whether the matched portion of the string is highlighted with the negative VT sequence.</value>
+        private bool Emphasize { get; set; }
+
+        /// <summary>
+        /// Stores the starting index of each match within the line.
+        /// </summary>
+        private readonly List<int> _matchIndexes;
+
+        /// <summary>
+        /// Stores the length of each match within the line.
+        /// </summary>
+        private readonly List<int> _matchLengths;
+
+        /// <summary>
+        /// Stores a values indicating whether or not the terminal supports VT.
+        /// </summary>
+        private readonly bool _supportsVirtualTerminal;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MatchInfo"/> class.
+        /// </summary>
+        public MatchInfo()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MatchInfo"/> class.
+        /// </summary>
+        /// <param name="matchIndexes">Sets the matchIndexes.</param>
+        /// <param name="matchLengths">Sets the matchLengths.</param>
+        /// <param name="emphasize">Used for implementing -Emphasize.</param>
+        /// <param name="vt">Sets a value indicating whether or not virtual terminal is supported.</param>
+        public MatchInfo(List<int> matchIndexes, List<int> matchLengths, bool emphasize, bool vt)
```

Comment:
The `emphasize` and `vt` parameters seem unneeded. Instead I think there should be two constructors:
- The 0-parameter constructor means there's no emphasis (could privately have an `_emphasize` variable set to `false` for clarity).
- The 2-parameter constructor that takes indices and lengths means there is emphasis (could privately have `_emphasize` set to `true` but could also just check `_matchIndexes != null && _matchLengths != null`.

Reply:
Sounds good. I'll add a check for whether VT is supported before the 2-parameter constructor is used since there's otherwise no way to check that within matchInfo.